docs/proposals/resource-quota-scoping.md

```diff
@@ -0,0 +1,362 @@
+the intersection of enumerated `scopes`.
+
+Adding a `scope` to a quota limits the number of resources
+it supports to those that pertain to the `scope`.  Specifying
+a resource on the quota object outside of the allowed set
+would result in a validation error.
+
+| Scope | Description |
+| ----- | ----------- |
+| Terminating | Match `kind=Pod` where `spec.activeDeadlineSeconds >= 0` |
+| NotTerminating | Match `kind=Pod` where `spec.activeDeadlineSeconds = nil` |
+| BestEffort | Match `kind=Pod` where `status.qualityOfService in (BestEffort)` |
+| NotBestEffort | Match `kind=Pod` where `status.qualityOfService not in (BestEffort)` |
+
+A `BestEffort` scope restricts a quota to tracking the following resources:
+
+* pod
+
+A `Terminating`, `NotTerminating`, `NotBestEffort` scope restricts a quota to
+tracking the following resources:
+
+* pod
+* memory, memory.request, memory.limit
+* cpu, cpu.request, cpu.limit
+
+## Data Model Impact
+
+```
+// The following identify resource constants for Kubernetes object types
+const (
+  // CPU Request, in cores
+  ResourceCPURequest ResourceName = "cpu.request"
+  // CPU Limit, in bytes
+  ResourceCPULimit ResourceName = "cpu.limit"
+  // Memory Request, in bytes
+  ResourceMemoryRequest ResourceName = "memory.request"
+  // Memory Limit, in bytes
+  ResourceMemoryLimit ResourceName = "memory.limit"
+)
+
+// A scope is a filter that matches an object
+type ResourceQuotaScope string
+const (
+  ResourceQuotaScopeTerminating ResourceQuotaScope = "Terminating"
+  ResourceQuotaScopeNotTerminating ResourceQuotaScope = "NotTerminating"
+  ResourceQuotaScopeBestEffort ResourceQuotaScope = "BestEffort"
+  ResourceQuotaScopeNotBestEffort ResourceQuotaScope = "NotBestEffort"
+)
+
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+// The quota matches by default on all objects in its namespace.
+// The quota can optionally match objects that satisfy a set of scopes.
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  Hard ResourceList `json:"hard,omitempty"`
+  // Scopes is the set of filters that must match an object for it to be
+  // tracked by the quota
+  Scopes []ResourceQuotaScope `json:"scopes,omitempty"`
+}
+```
+
+## Rest API Impact
+
+None.
+
+## Security Impact
+
+None.
+
+## End User Impact
+
+The `kubectl` commands that render quota should display its scopes.
+
+## Performance Impact
+
+This feature will make having more quota objects in a namespace
+more common in certain clusters.  This impacts the number of quota
+objects that need to be incremented during creation of an object
+in admission control.  It impacts the number of quota objects
+that need to be updated during controller loops.
+
+## Developer Impact
+
+None.
+
+## Alternatives
+
+This proposal initially enumerated a solution that leveraged a
+`FieldSelector` on a `ResourceQuota` object.  A `FieldSelector`
```

Comment:
Long term, I am in favor of FieldSelector to handle edge cases, and scopes to handle common cases.